test/src/unittests/highlevel/test_chromacrosssimilarity.py

```diff
@@ -0,0 +1,127 @@
+    spectralWhitening = estd.SpectralWhitening(maxFrequency= 3500,
+                                                sampleRate=44100)
+    hpcp = estd.HPCP(sampleRate=44100,
+                    maxFrequency=3500,
+                    minFrequency=100,
+                    referenceFrequency=440,
+                    nonLinear=True,
+                    harmonics=8,
+                    size=12)
+    #compute hpcp for each frame and add the results to the pool
+    for frame in frameGenerator:
+        spectrum_mag = spectrum(window(frame))
+        frequencies, magnitudes = spectralPeaks(spectrum_mag)
+        w_magnitudes = spectralWhitening(spectrum_mag,
+                                        frequencies,
+                                        magnitudes)
+        hpcp_vector = hpcp(frequencies, w_magnitudes)
+    return hpcp_vector
+
+
+def cross_recurrent_plot(input_x, input_y, tau=1, m=1, kappa=0.095):
+    """
+    Constructs the Cross Recurrent Plot of two audio feature vector
+    """
+    from sklearn.metrics.pairwise import euclidean_distances
+    pdistances = euclidean_distances(input_x, input_y)
+    #pdistances = resample_simmatrix(pdistances)
+    transposed_pdistances = pdistances.T
+    eph_x = np.percentile(pdistances, kappa*100, axis=1)
+    eph_y = np.percentile(transposed_pdistances, kappa*100, axis=1)
+    x = eph_x[:,None] - pdistances
+    y = eph_y[:,None] - transposed_pdistances
+    #apply heaviside function to the array (Binarize the array)
+    x = np.piecewise(x, [x<0, x>=0], [0,1])
+    y = np.piecewise(y, [y<0, y>=0], [0,1])
+    crp = x*y.T
+    return crp
+
+
+class TestChromaCrossSimilarity(TestCase):
+
+    query = estd.MonoLoader(filename=join(testdata.audio_dir, 'recorded', 'mozart_c_major_30sec.wav'))()
+    reference = estd.MonoLoader(filename=join(testdata.audio_dir, 'recorded', 'Vivaldi_Sonata_5_II_Allegro.wav'))
+    query_hpcp = hpcp(query)
+    reference_hpcp = hpcp(reference)
+    expected = cross_recurrent_plot(query_hpcp, reference_hpcp)
+
+    def testEmpty(self):
+        self.assertComputeFails(estd.CrossSimilarityMatrix(), [])
+
+    def testRegressionStandard(self):
+        """Tests standard ChromaCrossSimilarity algo"""
+        csm = estd.ChromaCrossSimilarity(oti=False, stackFrameSize=1)
+        result = csm.compute(self.query_hpcp, self.reference_hpcp)
+        self.assertAlmostEqual(np.mean(self.expected), np.mean(result))
+        self.assertAlmostEqualVector(self.expected, result)
+
+    def testInvalidParam(self):
+        self.assertConfigureFails(estd.ChromaCrossSimilarity(), { 'binarizePercentile': -1 })
+        self.assertConfigureFails(estd.ChromaCrossSimilarity(), { 'otiBinary': -1 })
+
+    def testOTIBinaryCompute(self):
+        """Tests standard ChromaCrossSimilarity algo when param 'otiBinary=True'"""
+        # test oti-based binary sim matirx method
+        self.assertComputeFails(estd.ChromaCrossSimilarity(otiBinary=True), [])
```

Comment:
There's only a fail test on empty input here, but no OTI regresssion.

Reply:
We don't have any reference implementation to test that against ours. And I did a python implementation before which is too slow to run. Should I use that?

Reply:
Yes, we can try to use that.